typing/ctype.ml

```diff
@@ -793,9 +793,7 @@ let rec generalize_expansive env var_level visited ty =
     Hashtbl.add visited ty.id ();
     match ty.desc with
       Tconstr (path, tyl, abbrev) ->
-        let variance =
-          try (Env.find_type path env).type_variance
-          with Not_found -> List.map (fun _ -> Variance.may_inv) tyl in
```

Comment:
So why is this useless?

Reply:
What if you have: `let x : _ foo = []`, foo comes from another module and is defined as `type 'a foo = 'a bar list` and `bar` comes from yet another module and you don't have that .cmi?
I think there are basically no tests for this kind of missing .cmi situation.

Reply:
I don't think that particular example is a problem here: we're looking at `foo`, not its expansion, and `foo` definitely is in the environment.
But:
1. I might very well wrong
2. the line of thought makes sense, and there might be a scenario that gets us into trouble. Someone who knows more will have to comment.

Ping @garrigue?

Reply:
Ah nevermind: I thought it recursed on the body of the type constructor, but no (only on the arguments).

Reply:
Thanks! :)

I added a testcase and a comment.